doc/source/train/train.rst

```diff
@@ -10,6 +10,7 @@ Ray Train: Scalable Model Training
     PyTorch Guide <getting-started-pytorch>
     PyTorch Lightning Guide <getting-started-pytorch-lightning>
     Hugging Face Transformers Guide <getting-started-transformers>
+    XGBoost Guide <getting-started-xgboost>
```

Comment:
add a link to this user guide from the old user guide and mention the deprecation.